pkg/api/profile.go

```diff
@@ -94,6 +94,10 @@ func (s *Server) startCPUProfile(w http.ResponseWriter, r *http.Request) {
 		HandleAPIError(w, r, http.StatusBadRequest, fmt.Errorf("missing path"))
 		return
 	}
+
+	s.Lock()
```

Comment:
You misused the lock which is the cluster-level config lock. The lock we should use is the localprofile-leveled.